Run tests on Appveyor to debug rplugin on Windows

After work done on https://github.com/neovim/neovim/pull/7706, https://github.com/neovim/neovim/pull/7734, https://github.com/neovim/neovim/pull/7738, I can't make a simple echo class with an `EchoHello` command work on Windows. The APIs are fine because testing on them directly, bypassing the host and `rplugin/node`,  works on all OS consistently. `:checkhealth` and `:UpdateRemotePlugins` are ok now (on Windows) and I see `rplugin.vim` with `EchoHello` registered from nodejs plugin I made it.

That leaves msgpack and the `RPC` api from both sides which I don't know how to debug/fix yet.
To get more debug info, the tests should run on Appveyor as well. We can reuse the same configuration from neovim repo.

NVIM_NODE_LOG_FILE

```
2017-12-20T18:15:31.532Z - debug: host.start

2017-12-20T18:16:06.580Z - debug: host.start

2017-12-20T18:16:06.612Z - info: handleRequest:  poll

2017-12-20T18:16:06.626Z - debug: request received:  poll

2017-12-20T18:16:06.629Z - info: handleNotification:  C:/Users/JanEdmund/AppData/Local/nvim/rplugin/node/hello.js:command:EchoHello

2017-12-20T18:16:06.630Z - debug: host.handlePlugin:  C:/Users/JanEdmund/AppData/Local/nvim/rplugin/node/hello.js:command:EchoHello

2017-12-20T18:16:06.766Z - info: [C] Error loading child ChildPlugin C

2017-12-20T18:16:06.789Z - info: [C] { Error: Cannot find module 'C'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.Module._load (module.js:417:25)
    at Module.require (module.js:497:17)
    at Object.sandboxRequire [as require] (C:\Users\JanEdmund\lib\node6\node_modules\neovim\lib\host\factory.js:79:46)
    at createPlugin (C:\Users\JanEdmund\lib\node6\node_modules\neovim\lib\host\factory.js:105:39)
    at Object.loadPlugin (C:\Users\JanEdmund\lib\node6\node_modules\neovim\lib\host\factory.js:139:16)
    at Host.getPlugin (C:\Users\JanEdmund\lib\node6\node_modules\neovim\lib\host\index.js:23:59)
    at Host.<anonymous> (C:\Users\JanEdmund\lib\node6\node_modules\neovim\lib\host\index.js:39:33)
    at next (native)
    at C:\Users\JanEdmund\lib\node6\node_modules\neovim\lib\host\index.js:7:71 code: 'MODULE_NOT_FOUND' }
```

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -25,7 +25,7 @@ build: off
 test_script:
   - yarn
   - yarn link
-  - npm install -g .
+  - npm install -g . # provider#node#Detect() checks bin/cli.js directly. So symlink does not work (Issue #50)
   - yarn run prep-integration-test
   - yarn test -- --coverage
 

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -25,6 +25,7 @@ build: off
 test_script:
   - yarn
   - yarn link
+  - npm install -g .
   - yarn run prep-integration-test
   - yarn test -- --coverage
 

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -1,10 +1,14 @@
 version: '{build}'
 clone_depth: 10
 environment:
-  NVIM_NODE_LOG_LEVEL: debug
+  matrix:
+    - NODE_VERSION: ""
+      NVIM_NODE_LOG_LEVEL: debug
+    - NODE_VERSION: "8"
+      NVIM_NODE_LOG_LEVEL: debug
 
 install:
-  - ps: Install-Product node 8
+  - ps: Install-Product node $Env:NODE_VERSION
   - node --version
   - npm --version
   - ps: |

--- a/.appveyor.yml
+++ b/.appveyor.yml
@@ -1,0 +1,40 @@
+version: '{build}'
+clone_depth: 10
+environment:
+  NVIM_NODE_LOG_LEVEL: debug
+
+install:
+  - ps: Install-Product node 8
+  - node --version
+  - npm --version
+  - ps: |
+      $zip = $Env:APPVEYOR_BUILD_FOLDER + '\nvim-win64.zip'
+      $nvim = $Env:APPVEYOR_BUILD_FOLDER + '\nvim-win64\'
+      $url = 'https://github.com/neovim/neovim/releases/download/nightly/nvim-win64.zip'
+      (New-Object Net.WebClient).DownloadFile($url, $zip)
+      [Reflection.Assembly]::LoadWithPartialName('System.IO.Compression.FileSystem') > $null
+      [System.IO.Compression.ZipFile]::ExtractToDirectory($zip, $nvim)
+      $Env:PATH = $Env:PATH + ';' + $nvim + '\Neovim\bin\'
+  - echo %PATH%
+  - nvim --version
+  - npm install -g yarn
+  - yarn global add codecov
+
+build: off
+
+test_script:
+  - yarn
+  - yarn link
+  - yarn run prep-integration-test
+  - yarn test -- --coverage
+
+after_test:
+  - codecov
+
+deploy: off
+
+notifications:
+  - provider: Email
+    on_build_status_changed: true
+    on_build_success: false
+    on_build_failure: false

--- a/README.md
+++ b/README.md
@@ -1,8 +1,8 @@
 # neovim-client
 
-| CI (Linux, macOS)                       | Coverage                               | npm                             | Gitter                      |
-|-----------------------------------------|----------------------------------------|---------------------------------|-----------------------------|
-| [![Build Status Badge][]][Build Status] | [![Coverage Badge][]][Coverage Report] | [![npm version][]][npm package] | [![Gitter Badge][]][Gitter] |
+| CI (Linux, macOS) | CI (Windows) | Coverage | npm | Gitter |
+|-------------------|--------------|----------|-----|--------|
+| [![Build Status Badge][]][Build Status] | [![Windows Build Status Badge][]][Windows Build Status] | [![Coverage Badge][]][Coverage Report] | [![npm version][]][npm package] | [![Gitter Badge][]][Gitter] |
 
 Currently tested for node >= 8
 
@@ -144,6 +144,8 @@ The tests and `scripts` can be consulted for more examples.
 
 [Build Status Badge]: https://travis-ci.org/neovim/node-client.svg?branch=master
 [Build Status]: https://travis-ci.org/neovim/node-client
+[Windows Build Status Badge]: https://ci.appveyor.com/api/projects/status/r989vwm83m7qc67m?svg=true
+[Windows Build Status]: https://ci.appveyor.com/project/rhysd/node-client
 [Coverage Badge]: https://codecov.io/gh/neovim/node-client/branch/master/graph/badge.svg
 [Coverage Report]: https://codecov.io/gh/neovim/node-client
 [npm version]: https://img.shields.io/npm/v/neovim.svg

--- a/package.json
+++ b/package.json
@@ -66,7 +66,7 @@
     "prepublishOnly": "npm run build",
     "test": "npm run build && jest --runInBand",
     "test-staged": "jest --bail --no-cache --findRelatedTests",
-    "prep-integration-test": "(cd __tests__/integration/rplugin/node/test && npm install)",
+    "prep-integration-test": "cd __tests__/integration/rplugin/node/test && npm install",
     "precommit": "lint-staged",
     "build": "tsc --pretty -p tsconfig.json",
     "watch": "tsc --pretty -p tsconfig.json --watch true"

--- a/src/api/Neovim.test.ts
+++ b/src/api/Neovim.test.ts
@@ -1,5 +1,6 @@
 /* eslint-env jest */
 import * as cp from 'child_process';
+import * as path from 'path';
 // eslint-disable-next-line import/no-extraneous-dependencies
 import * as which from 'which';
 import { attach } from '../attach';
@@ -65,8 +66,7 @@ describe('Neovim API', () => {
 
     it('can change current working directory', async () => {
       const initial = await nvim.call('getcwd', []);
-      const parts = initial.split('/');
-      const newCwd = parts.slice(0, parts.length - 1).join('/');
+      const newCwd = path.dirname(initial);
 
       nvim.dir = newCwd;
       expect(await nvim.call('getcwd', [])).toBe(newCwd);

